packages/blockchain-wallet-v4-frontend/src/scenes/Login/index.tsx

```diff
@@ -34,53 +33,37 @@ import WalletEnterEmailOrGuid from './Wallet/EnterEmailOrGuid'
 import EnterPasswordWallet from './Wallet/EnterPassword'
 import TwoFAWallet from './Wallet/TwoFA'
 
-class Login extends PureComponent<InjectedFormProps<{}, Props> & Props> {
-  componentDidMount() {
-    this.props.authActions.initializeLogin()
-    if (window?._SardineContext) {
-      window._SardineContext.updateConfig({
-        flow: 'LOGIN'
-      })
-    }
-    this.props.analyticsActions.trackEvent({
-      key: Analytics.LOGIN_VIEWED,
-      properties: {
-        device_origin: this.props?.productAuthMetadata?.platform || 'WEB',
-        originalTimestamp: new Date().toISOString()
-      }
-    })
-  }
-
-  setStep = (step: LoginSteps) => {
-    this.props.formActions.change(LOGIN_FORM, 'step', step)
+const Login = (props: Props) => {
```

Comment:
How about destrucuting props and sub props right here only once, in order to prevent using _props.x_ all along the component and destructuring every time you need to (examples: line 42, 65, 66, 79, 112)

Reply:
+1 ^

Reply:
I can't right now because it's contents are spread in line 136 of this file - I am first reducing the amount of code and putting everything as close to where it is used as possible before doing so